pkg/apis/iam/v1/handler.go

```diff
@@ -1246,6 +1332,31 @@ func (h *handler) fetchAggregationRoles(ctx context.Context, role *iamv1.GlobalR
 	return roles, nil
 }
 
+func (h *handler) fetchAggregationProjectRoles(ctx context.Context, projectRole *iamv1.ProjectRole) ([]*iamv1.ProjectRole, error) {
+	roles := make([]*iamv1.ProjectRole, 0)
+	if projectRole == nil || projectRole.Annotations == nil {
+		return roles, nil
+	}
+	if v := projectRole.Annotations[common.AnnotationAggregationRoles]; v != "" {
+		var roleNames []string
+		if err := json.Unmarshal([]byte(v), &roleNames); err == nil {
```

Comment:
This error should be returned to the caller first, otherwise, it may be swallowed.